### PR TITLE
Support locked problems

### PR DESCRIPTION
**Pull request by TsFreddie**
Added support for locked problems.
The regex is changed to match favorite, locked and state:
```
★ 🔒 ✔ [760] Find Anagram Mappings                                        Easy   (75.23 %)
```
The lock icon is grabbed from EmojiOne v2.0, please change it or add a license.

Please review the regex before merge.

### CODE REVIEW

**Comment by jdneo** [comment]
Hi @TsFreddie Thank you for the contribution! Looks nice!

Just interested about the intension of supporting locked problem. Could you please explain more about this feature to me?

**Comment by TsFreddie** [comment]
I should say paid problems.

leetcode-cli does not give you all the problems with `leetcode list -q L` even if you are a premium user of leetcode.

image

It would be nice to be able to submit those problems as well.

**Comment by jdneo** [comment]
@TsFreddie Oh, got your idea. That means: although the icon is a lock, but the paid user still can open the problem and try to solve it. Am I right?

I don't have a premium account so I was not aware of that. 😄

**Comment by TsFreddie** [comment]
Exactly. leetcode-cli isn't very descriptive about what it means by "locked".

**Comment by jdneo** [comment]
Good to know that. Thanks for pointing out. 👍 

I still have some stuff need to do. Will review the PR in detail ASAP!

File: src/commands/list.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { executeCommand } from "../utils/cpUtils";
 import { DialogType, promptForOpenOutputChannel } from "../utils/uiUtils";
 
 export interface IProblem {
+    favorate: boolean;
```
Comment (jdneo) [inline src/commands/list.ts]: typo: favorite

File: src/commands/list.ts [hunk 2]
```diff
@@ -19,19 +21,23 @@ export async function listProblems(channel: vscode.OutputChannel): Promise<IProb
         if (leetCodeManager.getStatus() === UserStatus.SignedOut) {
             return [];
         }
-        const result: string = await executeCommand(channel, "node", [leetCodeBinaryPath, "list", "-q", "L"]);
+        const leetCodeConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("leetcode");
+        const showLocked = leetCodeConfig.get<string>("showLocked");
```
Comment (jdneo) [inline src/commands/list.ts]: nit: add a type definition: `const showLocked: boolean = ...`

File: package.json [hunk 3]
```diff
@@ -129,6 +129,11 @@
             {
                 "title": "LeetCode",
                 "properties": {
+                    "leetcode.showLocked": {
+                        "type": "boolean",
+                        "default": true,
```
Comment (jdneo) [inline package.json]: Suggest we set the default value to false, since I think there are more common users than the premium ones.

**Comment by jdneo** [comment]
@TsFreddie Just some small changes, thanks.

**Comment by jdneo** [comment]
Great! Thanks.

**Comment by ismailhkose** [comment]
image

**Pull request merged**